### PR TITLE
TEST: @patch-triage bot fixture — do not merge

### PR DESCRIPTION
**Pull request by janzenisaac**
This PR is a test fixture for exercising the `@patch-triage` bot's `fix_and_commit` flow against a real PR on this repo. **Not for merge.**

## What's in the PR

A small new utility `DiscourseAi::Utils::TextSummary` with two methods (`truncate`, `word_count`), plus a spec.

## What's wrong with it

1. **Lint:** two orphan blank lines between members of the class (one after `DEFAULT_MAX_CHARS`, one between `truncate` and `word_count`). `stree write` would auto-fix both. Same shape that tripped CI on #39869.

2. **Behavior gap intentionally left for a reviewer to spot:** `truncate` doesn't strip trailing whitespace before appending the ellipsis, so `truncate("hello   world", max: 5)` returns `"hello  …"` with two trailing spaces baked in. The spec doesn't cover the boundary case (`text.length == max`) either.

## How to test the bot

I'll seed a review comment on `truncate` immediately after this opens, then mention `@patch-triage apply the suggestions` so the bot picks the work up via the `pr-fix` agent.

Expected behavior on the bot's side:
- Reads the review comment
- Adds `.rstrip` before the ellipsis on `truncate`
- Adds a spec for the `text.length == max` boundary case
- Lint pass (`stree write`) auto-fixes the two orphan blank lines
- Commits + pushes back to this branch
- Edits the placeholder comment with the result

After we verify the flow end-to-end, close + delete the branch.

### CODE REVIEW

File: plugins/discourse-ai/lib/utils/text_summary.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    class TextSummary
+      DEFAULT_MAX_CHARS = 1_000
+
+
+      # Returns the text truncated to `max` characters with an ellipsis
+      # appended, or the original text if it already fits.
+      #
+      # Returns nil for nil input so callers can chain `&.`.
+      def self.truncate(text, max: DEFAULT_MAX_CHARS)
+        return nil if text.nil?
```
Comment (janzenisaac) [inline plugins/discourse-ai/lib/utils/text_summary.rb:16]: Two suggestions on `truncate`:

1. **Strip trailing whitespace before appending the ellipsis.** Right now `truncate("hello   world", max: 5)` returns `"hello  …"` with two trailing spaces baked in — looks busted in the UI. Add a `.rstrip` before the interpolation.

2. **Add a spec for the boundary case** where `text.length == max`. The docstring says "the original text if it already fits" but nothing currently pins down that we don't truncate at the exact boundary.

Comment (janzenisaac) [reply plugins/discourse-ai/lib/utils/text_summary.rb:16]: @patch-triage apply the suggestions

Comment (janzenisaac) [reply plugins/discourse-ai/lib/utils/text_summary.rb:16]: @patch-triage apply the suggestions

Comment (janzenisaac) [reply plugins/discourse-ai/lib/utils/text_summary.rb:16]: @patch-triage apply the suggestions

Comment (janzenisaac) [reply plugins/discourse-ai/lib/utils/text_summary.rb:16]: @patch-triage apply the suggestions

**Comment by janzenisaac** [comment]
@patch-triage apply the suggestions
